### ISSUE TITLE
No icons appearing with latest Android User Agent

### ISSUE THREAD

**Issue by TomBonnot** [rignaneseleo/SlimSocial-for-Facebook#174]
Device : Google Pixel 7 
Version of SlimSocial : Latest 10.0.9
Custom User Agent : Mozilla/5.0 (Linux; Android 13) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.5672.76 Mobile Safari/537.36
Screenshot of the issue : 
![Screenshot_20230516-073621](https://github.com/rignaneseleo/SlimSocial-for-Facebook/assets/35734114/3e022fd1-3fb2-4f4e-ba3a-adc979b6542d)
Software that could interact or generate the bug : I have a DNS with Rethink DNS but I have totally excluded SlimSocial from it, so it shouldn’t interfere. 
Thank you for your time and your so useful app

**Comment by rignaneseleo** [comment]
I'm pretty sure it's a DNS problem. Be sure all the facebook domains are excluded. This app uses touch.facebook.com for example.

**Comment by sonofevil** [comment]
I am also having this issue. No changes to DNS or user agent. While loading the feed, the icons at the top appear at first, but then get replaced by boxes once the page is loaded. To me the boxes look like missing unicode symbol placeholders, which would suggest a font issue.

**Comment by sonofevil** [comment]
Interesting: With an iPhone user agent the icons are immediately broken, whereas with an Android agent there is some delay. The icons work fine with a desktop Linux agent, but the overall UI shrinks.

**Comment by FadeDave77** [comment]
If you turn off dark mode in the app settings, the icons return. There should be a toggle or something for dark mode on the main fb page and for everywhere else.

**Comment by sonofevil** [comment]
Is there some way at all to remote debug the app's css via adb like there is for Firefox for Android? I suspect it might be as easy to fix as changing the font in the css.

**Comment by sonofevil** [comment]
Right, so the buttons use character codes that are not in Unicode and likely only supported by proprietary fonts. On Firefox for Android, the font family it falls back on is "popular-symbols", a font I can't even find on Google. This is the fallback list in the css:

`.dfr .native-text { font-family: Roboto-V2, popular-symbols, lite-glyphs-outlined, lite-glyphs-filled, snaptu-symbols, NotoSansDevanagari, NotoSansMalayalam, NotoSansGujarati, NotoSansKannada, NotoSansGurmukhi, NotoSansSinhala, BengaliOTS, RobotoThaiLao, NotoSansHebrew, NotoSansKhmer, NotoSansTamil, NotoSansTelugu, NotoSansArmenian, NotoSansGeorgian, NotoSansEthiopic, NotoSansOriya, NotoSansTibetan, NotoSansCherokee, NotoSansArabic, NotoSansMyanmar, system-ui, Arial, sans-serif }`

Of the others, the only one that works is "lite-glyphs-outlined", also not found on Google, the rest displays placeholders. The fonts in question seem to be loaded via .woff2 file from fbcdn.net.

Unfortunately the custom css option in the settings seems to be entirely non-functional, otherwise I could try to force those fonts.

**Comment by sonofevil** [comment]
Could this be the culprit? https://github.com/rignaneseleo/SlimSocial-for-Facebook/blob/82d63bc696a924cc48d27a522679a7145248f7bc/SlimSocial_for_Facebook/lib/utils/css.dart#L286

It would be strange because FB's specific element styles should override the * selector. There is no other instance of font-family being used in the app's css. Perhaps the app fails to cache the .woff2 file correctly?

**Comment by rignaneseleo** [comment]
Thank you for the report, and I am sorry about the long delay.

This report is against an old version. The app is in active development again since August 2026, and much of it changed: the user agent, a new "Use desktop site" setting, the feed filters, the Messenger screen and photo upload. Old reports are no longer a good reference.

Note: Missing icons with a custom user agent on 10.0.9. The user agent handling changed several times since then.

I close this issue to clean up the tracker. Please install **26.09.03+128** or later, from the Releases page, the Play Store or F-Droid. If the problem is still there, open a new issue with your device model, your Android version and the app version.
